recipes/gdcm/all/conanfile.py

```diff
@@ -0,0 +1,224 @@
+            set(GDCM_LIBRARY_DIRS ${{GDCM_LIB_ROOT}})
+        """.format(v_major=v.major,
+                   v_minor=v.minor,
+                   v_patch=v.patch,
+                   build_shared_libs="ON" if self.options.shared else "OFF",
+                   gdcm_subdir=self._gdcm_subdir))
+        tools.save(variables_file, content)
+
+    @staticmethod
+    def _create_cmake_module_alias_targets(targets_file, libraries):
+        content = "\n".join([textwrap.dedent("""\
+            if(TARGET {aliased} AND NOT TARGET {alias})
+                add_library({alias} INTERFACE IMPORTED)
+                set_property(TARGET {alias} PROPERTY INTERFACE_LINK_LIBRARIES {aliased})
+            endif()
+            """.format(alias=library, aliased="GDCM::"+library)) for library in libraries])
+        tools.save(targets_file, content)
+
+    @property 
+    def _gdcm_subdir(self):
+        v = tools.Version(self.version)
+        return "gdcm-{}.{}".format(v.major, v.minor)
+
+    @property
+    def _gdcm_builddir(self):
+        return os.path.join("lib", self._gdcm_subdir)
+
+    @property
+    def _gdcm_cmake_module_aliases_path(self):
+        return os.path.join("lib", self._gdcm_subdir, "conan-official-gdcm-targets.cmake")
+
+    @property
+    def _gdcm_cmake_variables_path(self):
+        return os.path.join("lib", self._gdcm_subdir, "conan-official-gdcm-variables.cmake")
+
+    @property
+    def _gdcm_build_modules(self):
+        return [self._gdcm_cmake_module_aliases_path, self._gdcm_cmake_variables_path]
+
+    @property
+    def _gdcm_libraries(self):
+        gdcm_libs = ["gdcmcharls",
+                     "gdcmCommon",
+                     "gdcmDICT",
+                     "gdcmDSED",
+                     "gdcmIOD",
+                     "gdcmjpeg12",
+                     "gdcmjpeg16",
+                     "gdcmjpeg8",
+                     "gdcmMEXD",
+                     "gdcmMSFF",
+                     "socketxx"]
+        if self.settings.os == "Windows":
+            gdcm_libs.append("gdcmgetopt")
+        else:
+            gdcm_libs.append("gdcmuuid")
+        return gdcm_libs
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "GDCM"
+        self.cpp_info.names["cmake_find_multi_package"] = "GDCM"
+        gdcm_libs = self._gdcm_libraries
+        for lib in gdcm_libs:
+            self.cpp_info.components[lib].libs = [lib]
+            self.cpp_info.components[lib].includedirs = [os.path.join("include", self._gdcm_subdir)]
+            self.cpp_info.components[lib].builddirs = [self._gdcm_builddir] 
+            self.cpp_info.components[lib].build_modules["cmake"] = self._gdcm_build_modules
+            self.cpp_info.components[lib].build_modules["cmake_find_package"] = self._gdcm_build_modules
+            self.cpp_info.components[lib].build_modules["cmake_find_multi_package"] = self._gdcm_build_modules
+            self.cpp_info.components[lib].names["cmake_find_package"] = lib
```

Comment:
are explicit names needed if they are the same as component name?

Reply:
Since the names of each target are unlikely to collide... this is a great question 
... I believe you can remove the extra name... default is the component